lib/cli/test/unit/Cardano/CLISpec.hs

```diff
@@ -363,6 +364,17 @@ spec = do
         textRoundtrip $ Proxy @(Port "test")
         textRoundtrip $ Proxy @MnemonicSize
 
+    describe "Transaction ID decoding from text" $ do
+
+        it "Should produce a user-friendly error message on failing to \
+            \decode a transaction ID." $ do
+
+            let err = TextDecodingError
+                    "A transaction ID should be a hex-encoded string of 64 \
+                    \characters."
+
+            fromText @TxId "not-a-transaction-id" `shouldBe` Left err
```

Comment:
I would prefer  `` `shouldSatisfy` isLeft`` or `isTextDecodingError` rather than checking that the error string is exactly the same as defined in the other file. Well of course it is!

Reply:
> I would prefer `shouldSatisfy` isLeft or isTextDecodingError rather than checking that the error string is exactly the same as defined in the other file. Well of course it is!

Well, the point of _this_ particular test is to verify that the CLI reports the _user-friendly string_ rather than _some other string_. (The CLI originally reported a rather non-specific string defined in `Cardano.Wallet.Primitive.Types`.)

As for quoting the string directly, this is the style that's prevalent throughout our tests. I'd definitely be in favour of revising it in some way (for example, defining these strings just once), but perhaps we should debate that in the channel?